design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+
+---
+Wireframes
+
+[Filtering Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.19&scaling=min-zoom&starting-point-node-id=7379%3A82944&show-proto-sidebar=1)
+
+---
+
+### 1.3 Editing Existing List Data
+
+Users can single-click a List data cell or click the `show all` button to expand the selected the data cell and enter editing mode. A background shade is added to each List item, not the delimiters, to highlight the nature of List data type. Under cell editing mode, users can again single-click the item that they wish to edit to finish the edit without compromising the data structure.
+
+To add a new List item in the cell, users can simply single-click the white space within the cell when under editing mode. A new item placeholder and a new delimiter will show up. A preset Null value will be entered automatically if users have allowed and configured it.
+
+---
+Wireframes
+
+[Editing Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.24&scaling=min-zoom&starting-point-node-id=7809%3A84002&show-proto-sidebar=1)
+
+---
+
+### 1.4 Editing Existing List Data – Item Type Warning
+
+While editing List data items, if Mathesar detected that users were entering a potentially non-conforming item type (e.g., entering a number into a List of Text items), a warning modal would show up to inform users. 
+
+If Mathesar couldn't conform the new item to the existing List item data type, a serious warning would show up before converting all List data item in the column to Text.
+
+---
+Wireframes
+
+[Item Type Warning](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.22&scaling=min-zoom&starting-point-node-id=8268%3A90606&show-proto-sidebar=1)
+
+[All Column Item Change Warning](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.4&scaling=min-zoom&starting-point-node-id=8268%3A90982&show-proto-sidebar=1)
+
+---
+
+## 2. Converting Existing List Data to a Different Type
+
+Users can perform numerous data type operations to List data. There are two main categories of conversion:
+
+1. Converting List data to a basic data type as a whole.
+2. Converting List items to a different basic data type.
+
+### 2.1. Converting List Data to a Basic Type by Turning Off `Allow List`
+
+By default, the `Allow List` toggle in the column header menu would be turned on. While users are editing data cells, if Mathesar detects a comma delimiter and a data structure that resembles a List, Mathesar can make automatically inference and convert the column into a List data type. Users can still maintain the same basic data type for each List item, and the common basic data type settings, such as formatting and setting up Null values, are still available for List items.
```

Comment:
A couple of points here:

- We cannot turn on the "Allow List" toggle by default, since it changes how the data is stored in PostgreSQL. The user will need to turn this on.
- We will not be performing automatic data type conversions based on the data the user inputs, see my previous comment for more on that.